misk-hibernate/src/main/kotlin/misk/hibernate/TokenType.kt

```diff
@@ -0,0 +1,105 @@
+package misk.hibernate
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.usertype.UserType
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Types
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.primaryConstructor
+
+/**
+ * Binds any data class that wraps a String to a varchar in MySQL. This is most useful to enforce
+ * additional type-safety on identifier columns.
+ *
+ * To use, create a data class with a single non-null `String` property. The name of this property
+ * is not significant.
+ *
+ * ```
+ * data class MovieToken(val token: String)
+ * ```
+ *
+ * To use with comparison operators (ie. WHERE clauses with `<` or `>`) you must also implement
+ * `Comparable`:
+ *
+ * ```
+ * data class MovieToken(val token: String) : Comparable<MovieToken> {
+ *   override fun compareTo(other: MovieToken) = token.compareTo(other.token)
+ * }
+ * ```
+ */
+internal class TokenType<T : Any> private constructor(
```

Comment:
Since this is general purpose (rather than specific to tokens), maybe call it as it is: `BoxedStringType` or something.

Reply:
Fixed.